torchgeo/datasets/pastis.py

```diff
@@ -0,0 +1,453 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""PASTIS dataset."""
+
+import abc
+import os
+from collections.abc import Sequence
+from typing import Callable, Optional, Union
+
+import fiona
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.colors import ListedColormap
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import check_integrity, download_url, extract_archive
+
+
+class PASTIS(NonGeoDataset, abc.ABC):
+    """PASTIS dataset.
+
+    The `PASTIS <https://github.com/VSainteuf/pastis-benchmark>`__
+    dataset is a dataset for time-series panoptic segmentation of agricultural parcels.
+
+    Dataset features:
+
+    * support for the original PASTIS and PASTIS-R versions of the dataset
```

Comment:
How does one choose between PASTIS and PASTIS-R?

Reply:
PASTIS-R added the S1 data for each patch, see https://github.com/VSainteuf/pastis-benchmark/tree/main

Reply:
Not sure if we should document that or not. Depends on whether or not people will reasonably understand the difference before using.

Reply:
I think that is a safe assumption -- you have to understand the datasets you are using for any type of study. Put differently, you can't ever just use a dataset in a vacuum.